app/containers/TuneContainer/tests/saga.test.js

```diff
@@ -49,4 +58,54 @@ describe('TuneContainer saga tests', () => {
       })
     );
   });
+
+  it('should start task to watch for REQUEST_GET_TRACK_DETAILS', () => {
+    expect(generator.next().value).toEqual(takeLatest(tuneContainerTypes.REQUEST_GET_TRACK_DETAILS, getTrackDetails));
+  });
+
+  it('should ensure that the action SUCCESS_GET_TRACK_DETAILS is dispatched when the api call succeeds', () => {
+    getTrackDetailsGenerator = getTrackDetails({ songId });
+    getTrackDetailsGenerator.next().value;
+    const res2 = getTrackDetailsGenerator.next().value;
+
+    expect(res2).toEqual(call(getSongDetails, songId));
+
+    const data = { results: [{ songId }] };
+
+    expect(getTrackDetailsGenerator.next(apiResponseGenerator(true, data)).value).toEqual(
+      put({
+        type: tuneContainerTypes.SUCCESS_GET_TRACK_DETAILS,
+        data: data.results[0]
+      })
+    );
+  });
+
+  it('should ensure that the action FAILURE_GET_TRACK_DETAILS is dispatched when the api call succeeds', () => {
+    getTrackDetailsGenerator = getTrackDetails({ songId });
+    getTrackDetailsGenerator.next().value;
+    const res2 = getTrackDetailsGenerator.next().value;
+
+    expect(res2).toEqual(call(getSongDetails, songId));
+    const error = translate('something_went_wrong');
+
+    expect(getTrackDetailsGenerator.next(apiResponseGenerator(false, error)).value).toEqual(
+      put({
+        type: tuneContainerTypes.FAILURE_GET_TRACK_DETAILS,
+        error
+      })
+    );
+  });
+
+  it('should ensure that songsData is selected and SUCCESS_GET_TRACK_DETAILS is dispatched when songsData contains the trackId', () => {
+    getTrackDetailsGenerator = getTrackDetails({ songId, testData: songsData });
+    const res = getTrackDetailsGenerator.next().value;
+
+    const item = songsData.results.find((_song) => _song.trackId === songId);
```

Comment:
Don't duplicate this logic. I think we can test the same there right? Do we need to do this? The point of test is that even if we somehow write bad logic in saga, it should fail. Let's try to test the expression in saga.